### PR TITLE
xtensa/esp32: Support to select different clock source for RTC controller and close TODOs.

### PR DESCRIPTION
**Pull request by saramonteiro**
## Summary

This PR adds support for different clock sources for RTC controller and closes some pending TODOs from `esp32_rtc.c`.
These were:

- [x] It is missing the support for choosing the other source clocks to feed the slow_rtc_clk. 

- [x] It is missing to fix the overflow.

- [x] It is missing to change the delay function.

- [x] It is missing to improve the calibration routine 32K xtal oscilattor

## Impact

Most power management system and subsystems that use RTC Low­Power Clocks.

## Testing

Two tests were performed here:
- I ran `pmconfig` with  `esp32-devkitc:pm` defconfig and also   `esp32-devkitc:rtc` defconfig .
- I used a LA to analyse average frequency of RTC clocks.

### CODE REVIEW

File: arch/xtensa/src/esp32/Kconfig [hunk 1]
```diff
@@ -211,6 +211,40 @@ config ESP32_XTAL_26MHz
 
 endchoice # On-board Crystal Frequency
 
+choice ESP32_RTC_CLK_SRC
+    prompt "RTC clock source"
+    default ESP32_RTC_CLK_SRC_INT_RC
+    help
```
Comment (gustavonihei) [inline arch/xtensa/src/esp32/Kconfig:217]:
```suggestion
    ---help---
```
nit: minor styling issue

File: arch/xtensa/src/esp32/Kconfig [hunk 2]
```diff
@@ -211,6 +211,40 @@ config ESP32_XTAL_26MHz
...
+
+	config ESP32_RTC_CLK_SRC_INT_RC
+		bool "Internal 150kHz RC oscillator"
+	config ESP32_RTC_CLK_SRC_EXT_XTAL
+		bool "External 32kHz crystal"
+		select ESP_SYSTEM_RTC_EXT_XTAL
+	config ESP32_RTC_CLK_SRC_EXT_OSC
+		bool "External 32kHz oscillator at 32K_XN pin"
+	config ESP32_RTC_CLK_SRC_INT_8MD256
+		bool "Internal 8.5MHz oscillator, divided by 256 (~33kHz)"
```
Comment (gustavonihei) [inline arch/xtensa/src/esp32/Kconfig:245]: nit: A blank line is usually added between the configs

Comment (saramonteiro) [reply arch/xtensa/src/esp32/Kconfig]: done

File: arch/xtensa/src/esp32/Kconfig (hunk 1 above)
Comment (saramonteiro) [reply arch/xtensa/src/esp32/Kconfig]: done

**Review by acassis — APPROVED** [review]

**Pull request merged**